metaflow/task.py

```diff
@@ -493,6 +511,36 @@ def run_step(
                     )
                 )
 
+            # Add runtime dag info - for a nested foreach this may look like:
+            # foreach_indices: [0, 1]
+            # foreach_indices_truncated: [0]
+            # foreach_step_names: ['step1', 'step2']
+            foreach_indices, foreach_indices_truncated, foreach_step_names = (
+                self._dynamic_runtime_metadata(foreach_stack)
+            )
+            metadata.extend(
+                [
+                    MetaDatum(
+                        field="foreach-indices",
+                        value=foreach_indices,
+                        type="foreach-indices",
+                        tags=metadata_tags,
+                    ),
+                    MetaDatum(
+                        field="foreach-indices-truncated",
+                        value=foreach_indices_truncated,
+                        type="foreach-indices-truncated",
+                        tags=metadata_tags,
+                    ),
+                    MetaDatum(
```

Comment:
I believe this is only used in the siblings thing. If that's the case, we may be able to get rid of this when we refactor the siblings thing (if we do that). I am also a little confused as to why this is needed.

Reply:
Will refactor siblings function mostly to return siblings irrespective of whether it is in a for each or not.